server/remote/remote.go

```diff
@@ -76,6 +77,9 @@ type Remote interface {
 	// Branches returns the names of all branches for the named repository.
 	Branches(ctx context.Context, u *model.User, r *model.Repo) ([]string, error)
 
+	// BranchCommit returns sha of commit on top of the specified branch
+	BranchCommit(ctx context.Context, u *model.User, r *model.Repo, branch string) (string, error)
```

Comment:
Maybe `BranchLastCommitSha`?

Reply:
I think the func description does it ... if you think this is important go ahead and rename it ;)